packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -1197,6 +1197,30 @@ export function getTypeName(schema: Schema, usage?: SchemaContext[]): string {
   return getPriorityName(schema, usage) ?? schema.type ?? "any";
 }
 
+export function getSerializeTypeName(
+  schema: Schema,
+  usage?: SchemaContext[]
+): string {
+  const typeName = getTypeName(schema, usage);
+  const formattedName = typeName.replace("Date | string", "string");
+  const allTypes = formattedName.includes(" | ")
+    ? formattedName.split(" | ")
+    : formattedName.split("|");
+  const canSerialize = allTypes.every((type) => {
+    return (
+      ["string", "number", "boolean"].includes(type) ||
+      (type.startsWith('"') && type.endsWith('"')) ||
+      /^[0-9]*$/.test(type) ||
+      ["true", "false"].includes(type) ||
```

Comment:
I was wondering if we could leverage the isConstant to identify the literal values. Because identifying the literal type is error prone e.g do we consider negative number or float? do we consider the single quote string or three quote strings?

https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/unit/transform/transformSchemas.spec.ts#L77-L168

Reply:
I am using !isNaN(Number(type)) to determine if it's a number now. We probably can't use isConstant as the schema is an union, and we